src/redefined.resolver.ts

```diff
@@ -23,11 +27,31 @@ export class RedefinedResolver implements Resolver {
         this.resolvers = options?.resolvers || RedefinedResolver.createDefaultResolvers();
     }
 
-    async resolve(domain: string, networks?: string[], options?: CustomResolverServiceOptions): Promise<Account[]> {
-        const customOptions = options || {};
-        return flatten(// @ts-ignore
-          await Promise.all(this.resolvers.map(it => it.resolve(domain, { throwErrorOnInvalidDomain: false, ...customOptions }, networks)))
-        ).filter(it => !networks || networks.includes(it.network) || it.network === "evm")
+    async resolve(domain: string, networks?: string[], options?: CustomResolverServiceOptions): Promise<ResolverResponse> {
+        const result = await Promise.allSettled(
+            this.resolvers.map(async resolver => {
+                try {
+                    return await resolver.resolve(domain, networks, options)
+                } catch (e: any) {
+                    throw Error(JSON.stringify({ vendor: resolver.vendor, error: e.message }))
+                }
+            })
+        );
+
+        const data: ResolverResponse = {
+            response: [],
+            errors: [],
+        }
+
+        result.forEach(it => {
+            if (it.status === "fulfilled") {
+                data.response.push(...it.value.filter(it => !networks || networks.includes(it.network) || it.network === "evm"))
```

Comment:
why do we need filtering here?

Reply:
Because we need to show filtered data by networks if exists
It is better to do filtering in one place than in each specific resolver

redefined - exception :)